### PR TITLE
fix(curriculum): create new step 18 for spreadsheet project.

### LINKED ISSUE #52987: Building a Spreadsheet - step 17: setting hyphenated attribute values as properties
### Describe the Issue

I haven't looked at much of the new curriculum but I assume the fact that you need to convert the hyphenated aria attribute value to camel case for the property has been explained at some point?

Should using `setAttribute` not also be a valid way of setting attributes?

Do we teach `setAttribute` anywhere?

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-functional-programming-by-building-a-spreadsheet/step-17

### Your code

```



```

### Expected behavior

- Explain how to set the aria attribute as a property if it has not been explained.
- Allow the use of  `setAttribute` for setting attributes.
- Teach the `setAttribute` method somewhere if it has not been taught.

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Vijaychandra-Govindapalle**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #52987

I have created a new step 18 for spreadsheet project to properly teach `el.attribute` syntax. And updated step 17 accordingly.

### CODE REVIEW

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/65ae458e23954c3469e0c209.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+assert.match(code, /input\.aria[lL]abel\s*=/);
+
+```
+
+Property names should follow camel case when updating hyphenated HTML attribute values.
+
+```js
+assert.match(code, /input\.ariaLabel\s*=/);
+
+```
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/65ae458e23954c3469e0c209.md:40]: I don't think we need this test.
We can just use the first one with some updates.

```suggestion
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/65ae458e23954c3469e0c209.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+
+# --hints--
+
+You should set an attribute using the syntax `el.attribute = value`.
+
+```js
+assert.match(code, /input\.aria[lL]abel\s*=/);
+
+```
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/65ae458e23954c3469e0c209.md:33]:
```suggestion
You should have an `input.ariaLabel`.

```js
assert.match(code, /input\.ariaLabel\s*=/);

```
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/65ae458e23954c3469e0c209.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+Property names should follow camel case when updating hyphenated HTML attribute values.
+
+```js
+assert.match(code, /input\.ariaLabel\s*=/);
+
+```
+
+Attribute `aria-label` for the `input` element should be set to the same value as the `id` attribute.
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/65ae458e23954c3469e0c209.md:42]:
```suggestion
You should assign `letter + number` to `input.ariaLabel`.
```

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/65ae458e23954c3469e0c209.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+Set the `aria-label` attribute for the `input` element to the same value as the `id` attribute.
+
+
+# --hints--
+
+You should have an `input.ariaLabel`.
+
+```js
+assert.match(code, /input\.ariaLabel\s*=/);
+
```
Comment (gikf) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/65ae458e23954c3469e0c209.md:32]:
````suggestion
```
````

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/65ae458e23954c3469e0c209.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+
+You should assign `letter + number` to `input.ariaLabel`.
+
+```js
+assert.match(code, /input\.ariaLabel\s*=\s*letter\s\+\snumber/);
+
```
Comment (gikf) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/65ae458e23954c3469e0c209.md:39]:
```suggestion
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/65ae458e23954c3469e0c209.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+```js
+assert.match(code, /input\.ariaLabel\s*=/);
+
+
+
+You should assign `letter + number` to `input.ariaLabel`.
+
+```js
+assert.match(code, /input\.ariaLabel\s*=\s*letter\s\+\snumber/);
```
Comment (gikf) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/65ae458e23954c3469e0c209.md:38]:
```suggestion
assert.match(code, /input\.ariaLabel\s*=\s*letter\s*\+\s*number/);
```

**Review by jdwilkin4 — APPROVED** [review]
the changes look good to me 👍

**Review by JoyShaheb — APPROVED** [review]

**Pull request merged**